### PR TITLE
only add parent package to toInstall once

### PR DESCRIPTION
**Pull request by deitch**
When a dependency was also a root, we listed the package twice as to install. It did not affect the graph at all, and when we actually try to install, we see that it already was installed. But it did mess up the output for debugging for `build`/`publish`, and made `show-packages` hard to parse.

### CODE REVIEW

File: pkg/apk/repo.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ func (p *PkgResolver) GetPackagesWithDependencies(packages []string) (toInstall
 			return nil, nil, fmt.Errorf("could not find package %s", pkgName)
 		}
 		// do not add it to toInstall, as we want to have it in the correct order with dependencies
-		dependenciesMap[pkgName] = pkgs[0]
+		dependenciesMap[pkgs[0].Name] = pkgs[0]
```
Comment (jonjohnsonjr) [inline pkg/apk/repo.go:249]: I see this is already like this, but why do we only care about the 0th result of ResolvePackage?

Comment (deitch) [reply pkg/apk/repo.go:249]: Because that's the highest matching one. ResolvePackage returns all of the packages that match in descending priority.

**Review by jonjohnsonjr — APPROVED** [review]
